core/vault/src/main/java/uk/gov/dwp/vault/domain/DecryptedValue.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.dwp.vault.domain;
+
+public class DecryptedValue {
+
+    private final char[] clearText;
+
+    public DecryptedValue(char[] clearText) {
+        this.clearText = clearText;
+    }
+
+    public boolean matches(DecryptedValue otherPassword) {
+        if (clearText.length != otherPassword.clearText.length) {
+            return false;
+        }
+
+        for (int index = 0; index < clearText.length; index++) {
+            if (clearText[index] != otherPassword.clearText[index]) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    public char[] getClearText() {
+        return clearText;
+    }
+
+    public void erase() {
+        erase(clearText);
+    }
+
+    private static void erase(char[] clearText) {
```

Comment:
Does this need to be `static`?

Reply:
In fact, I'm not sure it needs to be here. This was some layover code from something I was playing with originally. I might just remove it if it's not actually being used.

Reply:
I've kept the `erase()` method in the case where we decide to "clear" the array after having read and finished with it. Removed the static modifier.